library/ssl_tls.c

```diff
@@ -593,7 +593,8 @@ static int tls_prf_generic( mbedtls_md_type_t md_type,
 {
     size_t nb;
     size_t i, j, k, md_len;
-    unsigned char tmp[128];
+    unsigned char *tmp;
```

Comment:
This change is not outside the PR's scope.
This is because this PR introduces an externally defined label, which could be in any variety of lengths.
Specifically, in this PR, using EAP-TLS, the required length was, IIRC, 133 bytes.
I could have changed the buffer length to fit the size for EAP-TLS, however since this feature is *not* only for EAP-TLS, but as a generic feature to derive keys, the buffer length should be adjusted accordingly

Reply:
Thanks for the clarification :)

Reply:
ok about a seperate commit, but I don't understand your comment about a dynamic buffer for the label.
It is given as input parameter as string literal, and we use `strlen` to determine its length

Reply:
This function implements the pseudorandom function for TLS 1.2, there is a function called `tls1_prf` for TLS 1.1 and TLS 1.0 that uses a stack buffer very similarly to this one. We need to make the same changes there too. (There is one for SSLv3 too, but that ignores the label and one for PSA, but that doesn't have a stack buffer like this.)